diff --git a/config.c b/config.c
index d75f88ca0c..7a417eb17f 100644
--- a/config.c
+++ b/config.c
@@ -1868,7 +1868,7 @@ static struct config_set_element *configset_find_element(struct config_set *cs,
 	return found_entry;
 }
 
-static int configset_add_value(struct config_set *cs, const char *key, const char *value)
+int configset_add_value(struct config_set *cs, const char *key, const char *value)
 {
 	struct config_set_element *e;
 	struct string_list_item *si;
