diff --git a/csum-file.c b/csum-file.c
index 0f35fa5ee4..cee7bd5aeb 100644
--- a/csum-file.c
+++ b/csum-file.c
@@ -86,7 +86,7 @@ int finalize_hashfile(struct hashfile *f, unsigned char *result, unsigned int fl
 	return fd;
 }
 
-void hashwrite(struct hashfile *f, const void *buf, unsigned int count)
+void real_hashwrite(struct hashfile *f, const void *buf, unsigned int count)
 {
 	while (count) {
 		unsigned offset = f->offset;
