diff --git a/fast-import.c b/fast-import.c
index b8b65a801c..1d276fb229 100644
--- a/fast-import.c
+++ b/fast-import.c
@@ -704,7 +704,7 @@ static struct tree_content *dup_tree_content(struct tree_content *s)
 	return d;
 }
 
-static void start_packfile(void)
+static void real_start_packfile(void)
 {
 	struct strbuf tmp_file = STRBUF_INIT;
 	struct packed_git *p;
@@ -811,7 +811,7 @@ static int loosen_small_pack(const struct packed_git *p)
 	return run_command(&unpack);
 }
 
-static void end_packfile(void)
+static void real_end_packfile(void)
 {
 	static int running;
 
@@ -1612,7 +1612,7 @@ static int update_branch(struct branch *b)
 	return 0;
 }
 
-static void dump_branches(void)
+void dump_branches(void)
 {
 	unsigned int i;
 	struct branch *b;
@@ -2144,7 +2144,9 @@ static uintmax_t change_note_fanout(struct tree_entry *root,
  * Complain if the following character is not what is expected,
  * either a space or end of the string.
  */
-static uintmax_t parse_mark_ref(const char *p, char **endptr)
+static uintmax_t parse_mark_ref(const char *p, char **endptr);
+
+static uintmax_t real_parse_mark_ref(const char *p, char **endptr)
 {
 	uintmax_t mark;
 
@@ -2208,9 +2210,8 @@ static void file_change_m(const char *p, struct branch *b)
 	case S_IFDIR:
 	case S_IFGITLINK:
 		/* ok */
-		break;
 	default:
-		die("Corrupt mode: %s", command_buf.buf);
+		break;
 	}
 
 	if (*p == ':') {
