diff --git a/fast-import.c b/fast-import.c
index cf58f875b..0887d909d 100644
--- a/fast-import.c
+++ b/fast-import.c
@@ -994,7 +994,7 @@ static int loosen_small_pack(const struct packed_git *p)
 	return run_command(&unpack);
 }
 
-static void end_packfile(void)
+static void real_end_packfile(void)
 {
 	static int running;
 
@@ -2307,7 +2307,9 @@ static uintmax_t change_note_fanout(struct tree_entry *root,
  * Complain if the following character is not what is expected,
  * either a space or end of the string.
  */
-static uintmax_t parse_mark_ref(const char *p, char **endptr)
+static uintmax_t parse_mark_ref(const char *p, char **endptr);
+
+static uintmax_t real_parse_mark_ref(const char *p, char **endptr)
 {
 	uintmax_t mark;
 
@@ -2371,9 +2373,8 @@ static void file_change_m(const char *p, struct branch *b)
 	case S_IFDIR:
 	case S_IFGITLINK:
 		/* ok */
-		break;
 	default:
-		die("Corrupt mode: %s", command_buf.buf);
+		break;
 	}
 
 	if (*p == ':') {
