diff --git a/notes.c b/notes.c
index 5c6286257..268556b92 100644
--- a/notes.c
+++ b/notes.c
@@ -429,11 +429,6 @@ static void load_subtree(struct notes_tree *t, struct leaf_node *subtree,
 
 		if (path_len == 2 * (GIT_SHA1_RAWSZ - prefix_len)) {
 			/* This is potentially the remainder of the SHA-1 */
-
-			if (!S_ISREG(entry.mode))
-				/* notes must be blobs */
-				goto handle_non_note;
-
 			if (hex_to_bytes(object_oid.hash + prefix_len, entry.path,
 					 GIT_SHA1_RAWSZ - prefix_len))
 				goto handle_non_note; /* entry.path is not a SHA1 */
