diff --git a/object-file.c b/object-file.c
index c3d866a287..904149ce90 100644
--- a/object-file.c
+++ b/object-file.c
@@ -1954,7 +1954,7 @@ static int freshen_packed_object(const struct object_id *oid)
 	return 1;
 }
 
-int write_object_file_flags(const void *buf, unsigned long len,
+int real_write_object_file_flags(const void *buf, unsigned long len,
 			    const char *type, struct object_id *oid,
 			    unsigned flags)
 {
