diff --git a/packfile.c b/packfile.c
index 7cd45aa4b2..f956192f66 100644
--- a/packfile.c
+++ b/packfile.c
@@ -1769,7 +1769,7 @@ off_t nth_packed_object_offset(const struct packed_git *p, uint32_t n)
 	}
 }
 
-off_t find_pack_entry_one(const unsigned char *sha1,
+off_t real_find_pack_entry_one(const unsigned char *sha1,
 				  struct packed_git *p)
 {
 	const unsigned char *index = p->index_data;
