diff --git a/sha1_file.c b/sha1_file.c
index b60ae15f7..ffc176e34 100644
--- a/sha1_file.c
+++ b/sha1_file.c
@@ -2753,7 +2753,7 @@ off_t nth_packed_object_offset(const struct packed_git *p, uint32_t n)
 	}
 }
 
-off_t find_pack_entry_one(const unsigned char *sha1,
+off_t real_find_pack_entry_one(const unsigned char *sha1,
 				  struct packed_git *p)
 {
 	const uint32_t *level1_ofs = p->index_data;
