diff --git a/builtin/clone.c b/builtin/clone.c
index e77339c847..b1a9973b7c 100644
--- a/builtin/clone.c
+++ b/builtin/clone.c
@@ -1534,7 +1534,7 @@ int cmd_clone(int argc,
 	submodule_progress = transport->progress;
 
 	transport_unlock_pack(transport, 0);
-	transport_disconnect(transport);
+	err = transport_disconnect(transport);
 
 	if (option_dissociate) {
 		close_object_store(the_repository->objects);
@@ -1545,8 +1545,9 @@ int cmd_clone(int argc,
 		return 1;
 
 	junk_mode = JUNK_LEAVE_REPO;
-	err = checkout(submodule_progress, filter_submodules,
-		       ref_storage_format);
+	if (!err)
+		err = checkout(submodule_progress, filter_submodules,
+			       ref_storage_format);
 
 	free(remote_name);
 	strbuf_release(&reflog_msg);
diff --git a/builtin/fetch.c b/builtin/fetch.c
index 80a64d0d26..eddb6616e2 100644
--- a/builtin/fetch.c
+++ b/builtin/fetch.c
@@ -2068,7 +2068,7 @@ static int fetch_one(struct remote *remote, int argc, const char **argv,
 {
 	struct refspec rs = REFSPEC_INIT_FETCH;
 	int i;
-	int exit_code;
+	int exit_code, disconnect_code;
 	int maybe_prune_tags;
 	int remote_via_config = remote_is_configured(remote, 0);
 
@@ -2134,9 +2134,9 @@ static int fetch_one(struct remote *remote, int argc, const char **argv,
 	exit_code = do_fetch(gtransport, &rs, config);
 	sigchain_pop(SIGPIPE);
 	refspec_clear(&rs);
-	transport_disconnect(gtransport);
+	disconnect_code = transport_disconnect(gtransport);
 	gtransport = NULL;
-	return exit_code;
+	return exit_code || disconnect_code;
 }
 
 int cmd_fetch(int argc,
