diff --git a/builtin/clone.c b/builtin/clone.c
index fba3c9c508..2c9fc2e5a9 100644
--- a/builtin/clone.c
+++ b/builtin/clone.c
@@ -1614,7 +1614,7 @@ int cmd_clone(int argc,
 	submodule_progress = transport->progress;
 
 	transport_unlock_pack(transport, 0);
-	transport_disconnect(transport);
+	err = transport_disconnect(transport);
 
 	if (option_dissociate) {
 		odb_close(the_repository->objects);
@@ -1625,10 +1625,11 @@ int cmd_clone(int argc,
 		return 1;
 
 	junk_mode = JUNK_LEAVE_REPO;
-	err = checkout(submodule_progress,
-		       &filter_options,
-		       filter_submodules,
-		       ref_storage_format);
+	if (!err)
+		err = checkout(submodule_progress,
+			       &filter_options,
+			       filter_submodules,
+			       ref_storage_format);
 
 	list_objects_filter_release(&filter_options);
 
diff --git a/builtin/fetch.c b/builtin/fetch.c
index a22c319467..4d26a40387 100644
--- a/builtin/fetch.c
+++ b/builtin/fetch.c
@@ -2389,7 +2389,7 @@ static int fetch_one(struct remote *remote, int argc, const char **argv,
 {
 	struct refspec rs = REFSPEC_INIT_FETCH;
 	int i;
-	int exit_code;
+	int exit_code, disconnect_code;
 	int maybe_prune_tags;
 	int remote_via_config = remote_is_configured(remote, 0);
 
@@ -2455,9 +2455,9 @@ static int fetch_one(struct remote *remote, int argc, const char **argv,
 	exit_code = do_fetch(gtransport, &rs, config, filter_options);
 	sigchain_pop(SIGPIPE);
 	refspec_clear(&rs);
-	transport_disconnect(gtransport);
+	disconnect_code = transport_disconnect(gtransport);
 	gtransport = NULL;
-	return exit_code;
+	return exit_code || disconnect_code;
 }
 
 int cmd_fetch(int argc,
