diff --git a/csum-file.c b/csum-file.c
index 2131ee6b12..f31d551613 100644
--- a/csum-file.c
+++ b/csum-file.c
@@ -111,7 +111,8 @@ void discard_hashfile(struct hashfile *f)
 	free_hashfile(f);
 }
 
-void hashwrite(struct hashfile *f, const void *buf, unsigned int count)
+void real_hashwrite(struct hashfile *f, const void *buf, unsigned int count);
+void real_hashwrite(struct hashfile *f, const void *buf, unsigned int count)
 {
 	while (count) {
 		unsigned left = f->buffer_len - f->offset;
