diff --git a/compat/win32/fscache.c b/compat/win32/fscache.c
index e2a6fe272f..3021ad553a 100644
--- a/compat/win32/fscache.c
+++ b/compat/win32/fscache.c
@@ -77,6 +77,10 @@ struct fsentry {
 };
 #pragma GCC diagnostic pop
 
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wflexible-array-extensions"
+#endif
 struct heap_fsentry {
 	union {
 		struct fsentry ent;
@@ -580,7 +584,7 @@ void fscache_flush(void)
 int fscache_lstat(const char *filename, struct stat *st)
 {
 	int dirlen, base, len;
-	struct heap_fsentry key[2];
+	struct heap_fsentry key0, key1;
 	struct fsentry *fse;
 	struct fscache *cache = fscache_getcache();
 
@@ -598,9 +602,9 @@ int fscache_lstat(const char *filename, struct stat *st)
 	dirlen = base ? base - 1 : 0;
 
 	/* lookup entry for path + name in cache */
-	fsentry_init(&key[0].u.ent, NULL, filename, dirlen);
-	fsentry_init(&key[1].u.ent, &key[0].u.ent, filename + base, len - base);
-	fse = fscache_get(cache, &key[1].u.ent);
+	fsentry_init(&key0.u.ent, NULL, filename, dirlen);
+	fsentry_init(&key1.u.ent, &key0.u.ent, filename + base, len - base);
+	fse = fscache_get(cache, &key1.u.ent);
 	if (!fse) {
 		errno = ENOENT;
 		return -1;
@@ -643,7 +647,7 @@ int fscache_lstat(const char *filename, struct stat *st)
 int fscache_is_mount_point(struct strbuf *path)
 {
 	int dirlen, base, len;
-	struct heap_fsentry key[2];
+	struct heap_fsentry key0, key1;
 	struct fsentry *fse;
 	struct fscache *cache = fscache_getcache();
 
@@ -661,9 +665,9 @@ int fscache_is_mount_point(struct strbuf *path)
 	dirlen = base ? base - 1 : 0;
 
 	/* lookup entry for path + name in cache */
-	fsentry_init(&key[0].u.ent, NULL, path->buf, dirlen);
-	fsentry_init(&key[1].u.ent, &key[0].u.ent, path->buf + base, len - base);
-	fse = fscache_get(cache, &key[1].u.ent);
+	fsentry_init(&key0.u.ent, NULL, path->buf, dirlen);
+	fsentry_init(&key1.u.ent, &key0.u.ent, path->buf + base, len - base);
+	fse = fscache_get(cache, &key1.u.ent);
 	if (!fse)
 		return mingw_is_mount_point(path);
 	return fse->reparse_tag == IO_REPARSE_TAG_MOUNT_POINT;
