diff --git a/http.c b/http.c
index 18508dd7f4..1fb0a0afcd 100644
--- a/http.c
+++ b/http.c
@@ -65,7 +65,7 @@ static char *ssl_key_type;
 static char *ssl_capath;
 static char *curl_no_proxy;
 static char *ssl_pinnedkey;
-static char *ssl_cainfo;
+char *ssl_cainfo;
 static long curl_low_speed_limit = -1;
 static long curl_low_speed_time = -1;
 static int curl_ftp_no_epsv;
@@ -424,11 +424,13 @@ static int http_options(const char *var, const char *value,
 		curl_ssl_try = git_config_bool(var, value);
 		return 0;
 	}
+#ifndef _WIN32
 	if (!strcmp("http.sslbackend", var)) {
 		free(http_ssl_backend);
 		http_ssl_backend = xstrdup_or_null(value);
 		return 0;
 	}
+#endif
 
 	if (!strcmp("http.schannelcheckrevoke", var)) {
 		if (value && !strcmp(value, "best-effort")) {
@@ -1381,9 +1383,6 @@ void http_init(struct remote *remote, const char *url, int proactive_auth)
 		}
 	}
 
-	if (curl_global_init(CURL_GLOBAL_ALL) != CURLE_OK)
-		die("curl_global_init failed");
-
 #ifdef GIT_CURL_HAVE_GLOBAL_TRACE
 	{
 		const char *comp = getenv("GIT_TRACE_CURL_COMPONENTS");
@@ -1479,7 +1478,6 @@ void http_cleanup(void)
 	curl_easy_cleanup(curl_default);
 
 	curl_multi_cleanup(curlm);
-	curl_global_cleanup();
 
 	string_list_clear(&extra_http_headers, 0);
 
