diff --git a/http.c b/http.c
index 239421a983..8d1086ca4d 100644
--- a/http.c
+++ b/http.c
@@ -68,7 +68,7 @@ static const char *curl_no_proxy;
 #ifdef GIT_CURL_HAVE_CURLOPT_PINNEDPUBLICKEY
 static const char *ssl_pinnedkey;
 #endif
-static const char *ssl_cainfo;
+const char *ssl_cainfo;
 static long curl_low_speed_limit = -1;
 static long curl_low_speed_time = -1;
 static int curl_ftp_no_epsv;
@@ -406,11 +406,13 @@ static int http_options(const char *var, const char *value,
 		curl_ssl_try = git_config_bool(var, value);
 		return 0;
 	}
+#ifndef _WIN32
 	if (!strcmp("http.sslbackend", var)) {
 		free(http_ssl_backend);
 		http_ssl_backend = xstrdup_or_null(value);
 		return 0;
 	}
+#endif
 
 	if (!strcmp("http.schannelcheckrevoke", var)) {
 		if (value && !strcmp(value, "best-effort")) {
@@ -1300,9 +1302,6 @@ void http_init(struct remote *remote, const char *url, int proactive_auth)
 	}
 #endif
 
-	if (curl_global_init(CURL_GLOBAL_ALL) != CURLE_OK)
-		die("curl_global_init failed");
-
 	http_proactive_auth = proactive_auth;
 
 	if (remote && remote->http_proxy)
@@ -1391,7 +1390,6 @@ void http_cleanup(void)
 	curl_easy_cleanup(curl_default);
 
 	curl_multi_cleanup(curlm);
-	curl_global_cleanup();
 
 	string_list_clear(&extra_http_headers, 0);
 
