diff --git a/http.c b/http.c
index bf6889a30c..f7e2a585ff 100644
--- a/http.c
+++ b/http.c
@@ -64,7 +64,7 @@ static char *ssl_key_type;
 static char *ssl_capath;
 static char *curl_no_proxy;
 static char *ssl_pinnedkey;
-static char *ssl_cainfo;
+char *ssl_cainfo;
 static long curl_low_speed_limit = -1;
 static long curl_low_speed_time = -1;
 static int curl_ftp_no_epsv;
@@ -423,11 +423,13 @@ static int http_options(const char *var, const char *value,
 		curl_ssl_try = git_config_bool(var, value);
 		return 0;
 	}
+#ifndef _WIN32
 	if (!strcmp("http.sslbackend", var)) {
 		free(http_ssl_backend);
 		http_ssl_backend = xstrdup_or_null(value);
 		return 0;
 	}
+#endif
 
 	if (!strcmp("http.schannelcheckrevoke", var)) {
 		if (value && !strcmp(value, "best-effort")) {
@@ -1380,9 +1382,6 @@ void http_init(struct remote *remote, const char *url, int proactive_auth)
 		}
 	}
 
-	if (curl_global_init(CURL_GLOBAL_ALL) != CURLE_OK)
-		die("curl_global_init failed");
-
 	if (proactive_auth && http_proactive_auth == PROACTIVE_AUTH_NONE)
 		http_proactive_auth = PROACTIVE_AUTH_IF_CREDENTIALS;
 
@@ -1470,7 +1469,6 @@ void http_cleanup(void)
 	curl_easy_cleanup(curl_default);
 
 	curl_multi_cleanup(curlm);
-	curl_global_cleanup();
 
 	string_list_clear(&extra_http_headers, 0);
 
