diff --git a/compat/mingw.c b/compat/mingw.c
index 303ffaa19f..e87f407f95 100644
--- a/compat/mingw.c
+++ b/compat/mingw.c
@@ -1,30 +1,30 @@
-#include "../git-compat-util.h"
+#include "git-compat-util.h"
 #include "win32.h"
 #include <aclapi.h>
 #include <sddl.h>
 #include <conio.h>
 #include <wchar.h>
 #include <winioctl.h>
-#include "../strbuf.h"
-#include "../run-command.h"
-#include "../abspath.h"
-#include "../alloc.h"
+#include "strbuf.h"
+#include "run-command.h"
+#include "abspath.h"
+#include "alloc.h"
 #include "win32/exit-process.h"
 #include "win32/lazyload.h"
-#include "../config.h"
-#include "../environment.h"
-#include "../trace2.h"
-#include "../symlinks.h"
-#include "../wrapper.h"
+#include "config.h"
+#include "environment.h"
+#include "trace2.h"
+#include "symlinks.h"
+#include "wrapper.h"
 #include "dir.h"
 #include "gettext.h"
 #define SECURITY_WIN32
 #include <sspi.h>
-#include "../write-or-die.h"
-#include "../repository.h"
+#include "write-or-die.h"
+#include "repository.h"
 #include "win32/fscache.h"
-#include "../attr.h"
-#include "../string-list.h"
+#include "attr.h"
+#include "string-list.h"
 #include "win32/wsl.h"
 
 #define HCAST(type, handle) ((type)(intptr_t)handle)
@@ -32,7 +32,6 @@
 void open_in_gdb(void)
 {
 	static struct child_process cp = CHILD_PROCESS_INIT;
-	extern char *_pgmptr;
 
 	strvec_pushl(&cp.args, "mintty", "gdb", NULL);
 	strvec_pushf(&cp.args, "--pid=%d", getpid());
