diff --git a/object-file.c b/object-file.c
index 1644a0d985..24b821dafd 100644
--- a/object-file.c
+++ b/object-file.c
@@ -29,6 +29,12 @@
 #include "setup.h"
 #include "streaming.h"
 
+#define write_object_file real_write_object_file
+
+int real_write_object_file(struct odb_source *source, const void *buf, size_t len,
+			    enum object_type type, struct object_id *oid,
+			    struct object_id *compat_oid_in, unsigned flags);
+
 /* The maximum size for an object header. */
 #define MAX_HEADER_LEN 32
 
