diff --git a/packfile.c b/packfile.c
index 80e35f1032..990dbb7295 100644
--- a/packfile.c
+++ b/packfile.c
@@ -1990,8 +1990,10 @@ off_t nth_packed_object_offset(const struct packed_git *p, uint32_t n)
 	}
 }
 
-off_t find_pack_entry_one(const struct object_id *oid,
-			  struct packed_git *p)
+off_t real_find_pack_entry_one(const struct object_id *oid,
+			       struct packed_git *p);
+off_t real_find_pack_entry_one(const struct object_id *oid,
+			       struct packed_git *p)
 {
 	const unsigned char *index = p->index_data;
 	uint32_t result;
