diff --git a/run-command.c b/run-command.c
index dbfe9834bd..269dd8f7d9 100644
--- a/run-command.c
+++ b/run-command.c
@@ -286,7 +286,8 @@ char *git_shell_path(void)
 #endif
 }
 
-static const char **prepare_shell_cmd(struct strvec *out, const char **argv)
+const char **prepare_shell_cmd(struct strvec *out, const char **argv);
+const char **prepare_shell_cmd(struct strvec *out, const char **argv)
 {
 	if (!argv[0])
 		BUG("shell command is empty");
