diff --git a/run-command.c b/run-command.c
index f40df01c77..adc77a4370 100644
--- a/run-command.c
+++ b/run-command.c
@@ -740,7 +740,7 @@ int start_command(struct child_process *cmd)
 	trace2_child_start(cmd);
 	trace_run_command(cmd);
 
-	fflush(NULL);
+	//fflush(NULL);
 
 	if (cmd->close_object_store)
 		close_object_store(the_repository->objects);
@@ -1224,7 +1224,7 @@ int start_async(struct async *async)
 
 #ifdef NO_PTHREADS
 	/* Flush stdio before fork() to avoid cloning buffers */
-	fflush(NULL);
+	//fflush(NULL);
 
 	async->pid = fork();
 	if (async->pid < 0) {
