diff --git a/run-command.c b/run-command.c
index 5ec3a46dcc..9d10ed27b8 100644
--- a/run-command.c
+++ b/run-command.c
@@ -722,7 +722,7 @@ int start_command(struct child_process *cmd)
 	trace2_child_start(cmd);
 	trace_run_command(cmd);
 
-	fflush(NULL);
+	//fflush(NULL);
 
 	if (cmd->close_object_store)
 		close_object_store(the_repository->objects);
@@ -1211,7 +1211,7 @@ int start_async(struct async *async)
 
 #ifdef NO_PTHREADS
 	/* Flush stdio before fork() to avoid cloning buffers */
-	fflush(NULL);
+	//fflush(NULL);
 
 	async->pid = fork();
 	if (async->pid < 0) {
