diff --git a/run-command.c b/run-command.c
index 7e4e2e7fb66..fc4932219fe 100644
--- a/run-command.c
+++ b/run-command.c
@@ -285,7 +285,8 @@ char *git_shell_path(void)
 #endif
 }
 
-static const char **prepare_shell_cmd(struct strvec *out, const char **argv)
+const char **prepare_shell_cmd(struct strvec *out, const char **argv);
+const char **prepare_shell_cmd(struct strvec *out, const char **argv)
 {
 	if (!argv[0])
 		BUG("shell command is empty");
